pymc3/step_methods/hmc/trajectory.py

```diff
@@ -110,10 +116,19 @@ def get_theano_hamiltonian_functions(model_vars, shared, logpt, potential,
     H, q, dlogp = _theano_hamiltonian(model_vars, shared, logpt, potential)
     energy_function, p = _theano_energy_function(H, q, **theano_kwargs)
     if use_single_leapfrog:
-        leapfrog_integrator = _theano_single_leapfrog(H, q, p, H.dlogp(q), **theano_kwargs)
+        if integrator == "leapfrog":
```

Comment:
how about a global dict `INTEGRATORS` that links these names to the integrator function?